include/envoy/event/dispatcher.h

```diff
@@ -153,6 +153,17 @@ class Dispatcher {
    */
   enum class RunType { Block, NonBlock };
   virtual void run(RunType type) PURE;
+
+  /**
+   * Optionally sets a custom buffer factory for the dispatcher.  This factory may be used by new
+   * connections on connection creation.
+   */
+  virtual void setBufferFactory(Buffer::FactoryPtr&& factory) PURE;
```

Comment:
Do you think it's worth having a setter on the interface? Seems like the factory could just be passed to the constructor of DispatcherImpl and then we avoid leaking the abstraction for changing it?